examples/advanced/tidb-cluster.yaml

```diff
@@ -1,295 +1,872 @@
   #   requests:
-  #     storage: 30Gi
-  #   schedulerName: default-scheduler
-  #   config:
-  #     addr: 0.0.0.0:8250
-  #     gc: 7
-  #     heartbeat-interval: 2
-
-  ## Ref: TiCDC is a tool for replicating the incremental data of TiDB
+  #     cpu: 1000m
+  #     memory: 1Gi
+  #     storage: 1Gi
+  #   limits:
+  #     cpu: 2000m
+  #     memory: 2Gi
+  #   imagePullPolicy: IfNotPresent
+  #   imagePullSecrets: secretName
+  #   hostNetwork: false
+  #   serviceAccount: advanced-tidb-pump
+  #   priorityClassName: system-cluster-critical
+  #   schedulerName: tidb-scheduler
+  #   nodeSelector:
+  #     app.kubernetes.io/component: pump
+  #   annotations:
+  #     node.kubernetes.io/instance-type: some-vm-type
+  #   tolerations: {}
+  #   configUpdateStrategy: RollingUpdate
+  #   statefulSetUpdateStrategy: RollingUpdate
+  #   podSecurityContext: {}
+  #   env: []
+  #   additionalContainers: []
+  #   additionalVolumes: []
+  #   additionalVolumeMounts: []
+  #   terminationGracePeriodSeconds: 30s
+  #   # Ref: https://docs.pingcap.com/tidb/stable/tidb-binlog-configuration-file#pump
+  #   config: |
+  #     gc = 6
+  #     [storage]
+  #     [storage.kv]
+  #       compaction-total-size-multiplier = 8.0
+
+  ## TiCDC is a tool for replicating the incremental data of TiDB
   ## Ref: https://pingcap.com/docs/tidb-in-kubernetes/stable/deploy-ticdc/
   # ticdc:
   #   baseImage: pingcap/ticdc
+  #   version: "v4.0.8"
   #   replicas: 3
+  #   storageClassName: local-storage
+  #   requests:
+  #     cpu: 1000m
+  #     memory: 1Gi
+  #   limits:
+  #     cpu: 2000m
+  #     memory: 2Gi
+  #   imagePullPolicy: IfNotPresent
+  #   imagePullSecrets: secretName
+  #   hostNetwork: false
+  #   serviceAccount: advanced-tidb-ticdc
+  #   priorityClassName: system-cluster-critical
+  #   schedulerName: tidb-scheduler
+  #   nodeSelector:
+  #     app.kubernetes.io/component: ticdc
+  #   annotations:
+  #     node.kubernetes.io/instance-type: some-vm-type
+  #   tolerations: {}
+  #   configUpdateStrategy: RollingUpdate
+  #   statefulSetUpdateStrategy: RollingUpdate
+  #   podSecurityContext: {}
+  #   env: []
+  #   additionalContainers: []
+  #   additionalVolumes: []
+  #   additionalVolumeMounts: []
+  #   terminationGracePeriodSeconds: 30s
+  #   # Ref: https://docs.pingcap.com/tidb/stable/deploy-ticdc#add-ticdc-to-an-existing-tidb-cluster-using-binary-not-recommended
+  #   config:
+  #     timezone: UTC
+  #     gcTTL: 86400
+  #     logLevel: info
+  #     logFile: /dev/stderr
 
-
+  ## TiFlash is the columnar storage extension of TiKV
   ## Ref: https://pingcap.com/docs/tidb-in-kubernetes/stable/deploy-tiflash/
   # tiflash:
+  #   ###############################
+  #   # Basic TiFlash Configuration #
+  #   ###############################
   #   baseImage: pingcap/tiflash
-  #   maxFailoverCount: 3
+  #   version: "v4.0.8"
   #   replicas: 1
+  #   # requests:
+  #   #   cpu: 1000m
+  #   #   memory: 1Gi
+  #   #   storage: 1Gi
```

Comment:
Remove this since not used for TiFlash.

Reply:
OK. The resource requirements are so complex 😂